File: api/research/quick.py
"""
Quick research query endpoint
POST /api/research/quick
"""

from http.server import BaseHTTPRequestHandler
import json
import sys
import os

# Add parent directory to path to import lib
sys.path.insert(0, os.path.join(os.path.dirname(__file__), '../..'))

from lib.research_agent import run_research_agent
from lib.auth import validate_api_key, send_unauthorized_response


class handler(BaseHTTPRequestHandler):
    def do_POST(self):
        # Validate API key first
        if not validate_api_key(self):
            send_unauthorized_response(self)
            return
        
        try:
            # Read request body
            content_length = int(self.headers.get('Content-Length', 0))
            body = self.rfile.read(content_length).decode('utf-8')
            data = json.loads(body) if body else {}
            
            # Get query from body
            query = data.get('query') or data.get('input_as_text')
            
            if not query:
                self.send_error_response(
                    400,
                    'Either "query" or "input_as_text" is required in request body'
                )
                return
            
            # Check for OpenAI API key
            if not os.getenv('OPENAI_API_KEY'):
                self.send_error_response(
                    500,
                    'OPENAI_API_KEY environment variable is not set'
                )
                return
            
            # Run the agent and get JSON output directly
            result = run_research_agent(query, data.get('workflowId'))
            
            # Send success response
            self.send_response(200)
            self.send_header('Content-Type', 'application/json')
            self.send_header('Access-Control-Allow-Origin', '*')
            self.send_header('Access-Control-Allow-Methods', 'POST, OPTIONS')
            self.send_header('Access-Control-Allow-Headers', 'Content-Type, X-API-Key, Authorization')
            self.end_headers()
            
            # Return the agent's JSON output directly
            self.wfile.write(json.dumps(result).encode())
            
        except Exception as e:
            self.send_error_response(500, str(e))
    
    def do_OPTIONS(self):
        self.send_response(200)
        self.send_header('Access-Control-Allow-Origin', '*')
        self.send_header('Access-Control-Allow-Methods', 'POST, OPTIONS')
        self.send_header('Access-Control-Allow-Headers', 'Content-Type, X-API-Key, Authorization')
        self.end_headers()
        return
    
    def send_error_response(self, status_code, message):
        self.send_response(status_code)
        self.send_header('Content-Type', 'application/json')
        self.send_header('Access-Control-Allow-Origin', '*')
        self.end_headers()
        
        response = {
            'success': False,
            'error': message
        }
        
        self.wfile.write(json.dumps(response).encode())

File: api/research/quick.js
import { runResearchAgentWorkflow } from '../../lib/researchAgent.js';

/**
 * API endpoint for quick research queries (simplified interface)
 * POST /api/research/quick
 * Body: { query } or { input_as_text }
 */
export default async function handler(req, res) {
  // Enable CORS
  res.setHeader('Access-Control-Allow-Credentials', true);
  res.setHeader('Access-Control-Allow-Origin', '*');
  res.setHeader('Access-Control-Allow-Methods', 'GET,OPTIONS,PATCH,DELETE,POST,PUT');
  res.setHeader(
    'Access-Control-Allow-Headers',
    'X-CSRF-Token, X-Requested-With, Accept, Accept-Version, Content-Length, Content-MD5, Content-Type, Date, X-Api-Version'
  );

  if (req.method === 'OPTIONS') {
    res.status(200).end();
    return;
  }

  if (req.method !== 'POST') {
    return res.status(405).json({ 
      success: false,
      error: 'Method not allowed' 
    });
  }

  try {
    // Accept both 'query' and 'input_as_text' for flexibility
    const input_as_text = req.body.input_as_text || req.body.query;

    if (!input_as_text) {
      return res.status(400).json({
        success: false,
        error: 'Either "input_as_text" or "query" is required in request body'
      });
    }

    // Check for OpenAI API key
    if (!process.env.OPENAI_API_KEY) {
      return res.status(500).json({
        success: false,
        error: 'OPENAI_API_KEY environment variable is not set'
      });
    }

    // Run the research agent workflow
    const result = await runResearchAgentWorkflow({
      input_as_text,
      workflowId: req.body.workflowId
    });

    // Return simplified response for quick queries
    res.status(200).json({
      success: true,
      query: input_as_text,
      response: result.output_text,
      parsed: result.output_parsed,
      timestamp: new Date().toISOString()
    });
  } catch (error) {
    console.error('Error in quick research query:', error);
    
    res.status(500).json({
      success: false,
      error: error.message || 'An error occurred while processing your query',
      details: process.env.NODE_ENV === 'development' ? error.stack : undefined
    });
  }
}

